pkg/pillar/zfs/zfs.go

```diff
@@ -23,6 +23,27 @@ var (
 	zfsPath = []string{"/hostfs", "zfs"}
 )
 
+//CreateDataset creates an empty dataset
+func CreateDataset(log *base.LogObject, dataset string) (string, error) {
+	args := append(zfsPath, "create", "-p", dataset)
+	var err error
+	var stdoutStderr []byte
+	tries := 0
+	maxTries := 3
+	for {
+		stdoutStderr, err = base.Exec(log, vault.ZfsPath, args...).CombinedOutput()
+		if err == nil {
+			return string(stdoutStderr), nil
+		}
+		tries++
+		if tries > maxTries {
+			break
+		}
+		time.Sleep(time.Second)
```

Comment:
Didn't you add a separate microservice a while back which would notify when the dataset is created? Or was that only for the zvol? 

Trying to understand if this is to handle the same thing that one can get an EINPROGRESS/EAGAIN and need to retry, and if so whether we have any confidence that under load it is sufficient with 3 tries in 3 seconds.

Reply:
I removed this retry from here, it is not needed. Seems, we can hit errors with busy dataset on remove (DestroyDataset).